lib/devise/multi_email/models/authenticatable.rb

```diff
@@ -48,22 +48,33 @@ def email=(new_email)
       module ClassMethods
         def find_first_by_auth_conditions(tainted_conditions, opts = {})
           filtered_conditions = devise_parameter_filter.filter(tainted_conditions.dup)
-          email = filtered_conditions.delete(:email)
+          criteria = filtered_conditions.extract!(:email, :unconfirmed_email)
 
-          if email && email.is_a?(String)
+          if criteria.keys.any?
             conditions = filtered_conditions.to_h.merge(opts).
-              reverse_merge(multi_email_association.reflection.table_name => { email: email })
+              reverse_merge(build_conditions criteria)
 
             resource = joins(multi_email_association.name).find_by(conditions)
-            resource.current_login_email = email if resource.respond_to?(:current_login_email=)
+            resource.current_login_email = criteria.values.first if resource
             resource
           else
             super(tainted_conditions, opts)
           end
         end
 
         def find_by_email(email)
-          joins(multi_email_association.name).where(multi_email_association.reflection.table_name => { email: email.downcase }).first
+          joins(multi_email_association.name).where(build_conditions email: email).first
+        end
+
+        def build_conditions(criteria)
+          criteria = devise_parameter_filter.filter(criteria)
+          if criteria[:unconfirmed_email]
+            criteria.merge!(primary: false, created_at: nil, email: criteria.delete(:unconfirmed_email))
```

Comment:
This will find an `Email` record that is not confirmed and not marked as primary. Devise would normally search for the email address on the `unconfirmed_email` column, and since we join on the `Email` model, these criteria ensure you find an equivalent to what Devise would expect out-of-the-box.